### PR TITLE
Handle full SHAs in product specs

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Fixes https://github.com/web-platform-tests/wpt.fyi/issues/672 by re-using Full SHA parsing on the product SHA part.

### CODE REVIEW

File: shared/params.go [hunk 1]
```diff
@@ -201,17 +201,29 @@ func ParseSHAParam(r *http.Request) (runSHA string, err error) {
...
+		return sha, err
+	}
+	return sha[:10], nil
+}
+
 // ParseSHAParamFull parses and validates the 'sha' param for the request.
 // It returns "latest" by default (and in error cases).
 func ParseSHAParamFull(r *http.Request) (runSHA string, err error) {
 	// Get the SHA for the run being loaded (the first part of the path.)
 	runSHA = "latest"
```
Comment (CYBAI) [inline shared/params.go]: This variable seems unused now in `ParseSHAParamFull`?

Comment (lukebjerring) [reply shared/params.go]: Done.

File: shared/params.go [hunk 2]
```diff
@@ -201,17 +201,28 @@ func ParseSHAParam(r *http.Request) (runSHA string, err error) {
 	return sha[:10], nil
 }
 
+// ParseSHA validates the given 'sha' value, cropping it to 10 chars.
+// It returns "latest" by default. (and in error cases).
```
Comment (foolip) [inline shared/params.go]:
```suggestion
// It returns "latest" by default (and in error cases).
```

File: shared/params_test.go [hunk 3]
```diff
@@ -344,6 +337,18 @@ func TestParseProductSpec(t *testing.T) {
...
 }
 
+func TestParseProductSpec_FullSHA(t *testing.T) {
+	sha := "0123456789aaaaabbbbbcccccdddddeeeeefffff"
+	r := httptest.NewRequest("GET", "http://wpt.fyi/?product=chrome@"+sha, nil)
+	filters, err := ParseTestRunFilterParams(r)
+	assert.Nil(t, err)
+	products := filters.GetProductsOrDefault()
+	assert.Len(t, products, 1)
+	if len(products) > 0 {
```
Comment (foolip) [inline shared/params_test.go]: Isn't this condition redundant given the above assertion? Or does the test keep running on failed asserts?

Comment (lukebjerring) [reply shared/params_test.go]: Unfortunately no, Golang keeps going on a failed assert.

Comment (foolip) [reply shared/params_test.go]: Oh, weird :)

**Pull request merged**